src/lib/spinel/radio_spinel.cpp

```diff
@@ -670,6 +670,12 @@ void RadioSpinel::HandleValueIs(spinel_prop_key_t aKey, const uint8_t *aBuffer,
             break;
         }
     }
+#if OPENTHREAD_SPINEL_CONFIG_ENABLE_VENDOR_HOOK
+    else if (aKey >= SPINEL_PROP_VENDOR__BEGIN && aKey < SPINEL_PROP_VENDOR__END)
+    {
+        error = VendorGetPropertyHandler(aKey);
```

Comment:
Should we rename it to `HandleValueId()` to make it similar to the name here?

Reply:
BTW, how this method  gets the received spinel frame? We should pass the parameters `aBuffer` and  `aLength`?

Reply:
BTW, Just to mention. the `HandleValueIs()` is used on "notification" spinel frame received from RCP (so can be used for vendor-specific events emitted by RCP).

If we want to get a vendor specific spinel frame (on `RadioSpinel)`, I think we can already use `Get()` here and have it parse the response.

Reply:
Our use case requires exactly that, an unsolicited notification by the rcp to the host.
In the VendorGetPropertyHandler we have something like:
```
bool my_bool;
SuccessOrExit(error = Get(SPINEL_PROP_VENDOR_PROPERTY, SPINEL_DATATYPE_BOOL_S, &my_bool));
```

Reply:
Should we rename from `VendorGetPropertyHandler()` to `HandleValueIs()` to be consistent with its purpose?

Reply:
On the rename: I could, but I feel it is more a Get/Set property

We have this code, where the callback has no notice of the RadioSpinel, but the function called by it requires an instance.
```
otError RadioSpinel::VendorGetPropertyHandler(spinel_prop_key_t aPropKey)
{
    ...
    case SPINEL_PROP_VENDOR_CALLBACK:
         bool mybool;
        SuccessOrExit(error = Get(SPINEL_PROP_VENDOR_CALLBACK, SPINEL_DATATYPE_BOOL_S, &mybool));
        if(mybool)
        {
             RegisterCallback(my_callback);
        }
    ...
}

void RadioSpinel::my_function(void)
{
    SuccessOrExit(error = Set(SPINEL_PROP_VENDOR_MY_KEY,....);
}

void my_callback(void)
{
    GetSpinelRadio.my_function(void)
}
```

Reply:
The reason I am suggesting this is this code path is for processing a "notification" (event) spinel frame sent by RCP and it is not a response by RCP to a "get prop" command (even though the spinel format is similar and same code can be used to parse the content for both cases).